code/modules/reagents/reagent_containers/food/cans.dm

```diff
@@ -44,28 +50,31 @@
 	if(istype(W, /obj/item/steelwool))
 		if(is_open_container())
 			switch(fuselength)
-				if(0 to 9)
-					user.visible_message("<b>[user]</b> stuffs some steel wool into \the [name].", SPAN_NOTICE("You feed steel wool into \the [name]. It will last approximately 10 seconds."))
-					fuselength = 10
+				if(-INFINITY to FUSELENGTH_MAX)
+					user.visible_message("<b>[user]</b> stuffs some steel wool into \the [name].", SPAN_NOTICE("You feed steel wool into \the [name], ruining it in the process. It will last approximately 10 seconds."))
+					fuselength = FUSELENGTH_MAX
 					update_icon()
+					qdel(W)
 					desc += " It has some steel wool stuffed into the opening."
-				if(10)
+					if(W.isFlameSource())
+						light_fuse(W, user, TRUE)
```

Comment:
You just qdel'd W on line 57 though? We can't call procs on it now.

Reply:
Made a mistake when changing the order of the lines, qdel(W) should've been after the flamesource check, will be fixed in the next commit.